src/imx_defaults.c

```diff
@@ -0,0 +1,329 @@
+#include "imx_defaults.h"
+
+
+
+
+// Return 0 if not valid.  Zero is not a valid platform type.
+int imxPlatformConfigTypeValid(uint32_t platformConfig)
+{
+    uint32_t platformType = platformConfig&PLATFORM_CFG_TYPE_MASK;
+    return (platformType > 0) && (platformType < PLATFORM_CFG_TYPE_COUNT);
```

Comment:
I can't see these macros in the PR, but is this right?  the platformType being < PLATFORM_CFG_TYPE_COUNT only applies it the TYPE MASK is in the lowest order bits (0-n), otherwise it may never be true.

Reply:
This code just ensures that the platformType which is a subset in the platformConfig is greater than 0 and less than the max value for platformType.  Looks correct to me.